LocalStorage/LocalStorageHelpers/LocalStorageProtocol.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  LocalStorageHelperProtocol.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 06/04/2024.
+//
+
+import Foundation
+import Combine
+
+protocol LocalStorageProtocol {
+    
+    /// outputs
+    var cities: PassthroughSubject<Result<[LocalStorageCity], LocalStorageError>, Never> {get}
+    
+    func getCitiesData() -> [LocalStorageCity]
+    
+    /// inputs
+    func addCityWeatherInfoToLocalStorage(weatherInfo: FormattedCityWeatherModel) throws
+    func deleteCityFromLocalStorage(_ cityName: String) throws
+    func clearDatabase() throws
```

Comment:
As you can see, there're three different models that the local storage can interact with. Ideally, you should interact with one model type.